standalone-examples/src/main/java/io/pravega/example/consolerw/ConsoleReader.java

```diff
@@ -30,67 +45,229 @@
+     */
+    private void readBasedOnStreamCuts(ReaderGroupConfig config) {
+        final String readerGroup = UUID.randomUUID().toString().replace("-", "");
         try (ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scope, controllerURI)) {
-            readerGroupManager.createReaderGroup(readerGroup, readerGroupConfig, Collections.singleton(streamName));
+            // Create a reader group using the configuration with the defined StreamCut boundaries.
+            readerGroupManager.createReaderGroup(readerGroup, config);
         }
 
         try (ClientFactory clientFactory = ClientFactory.withScope(scope, controllerURI);
-             EventStreamReader<String> reader = clientFactory.createReader("reader",
-                                                                           readerGroup,
-                                                                           new JavaSerializer<String>(),
-                                                                           ReaderConfig.builder().build())) {
-            System.out.format("******** Reading events from %s/%s%n", scope, streamName);
-            EventRead<String> event = null;
-            do {
-                try {
-                    event = reader.readNextEvent(READER_TIMEOUT_MS);
-                    if(event != null) {
-                        System.out.format("'%s'%n", event.getEvent());
+             EventStreamReader<String> reader = clientFactory.createReader("streamcut-reader",
+                     readerGroup, new JavaSerializer<>(), ReaderConfig.builder().build())) {
+
+            // The reader(s) will only read and display events within the StreamCut boundaries defined.
+            output("StreamCuts: Bounded processing example in stream %s/%s%n", scope, streamName);
+            output("Starting boundary for readers: %s (UnboundedStreamCut represents the head of the stream)%n",
+                    config.getStartingStreamCuts().get(Stream.of(scope, streamName)));
+            output("Terminal boundary for readers: %s (UnboundedStreamCut represents the tail of the stream)%n",
+                    config.getEndingStreamCuts().get(Stream.of(scope, streamName)));
+
+            EventRead<String> event;
+            try {
+                do {
+                    event = reader.readNextEvent(1000);
+                    if (event.getEvent() != null) {
+                        System.out.format("'%s'\n", event.getEvent());
                     }
-                } catch (ReinitializationRequiredException e) {
-                    //There are certain circumstances where the reader needs to be reinitialized
-                    e.printStackTrace();
-                }
-            }while(true);
+                } while (event.getEvent() != null);
+
+            } catch (ReinitializationRequiredException e) {
+                // There are certain circumstances where the reader needs to be reinitialized.
+                e.printStackTrace();
```

Comment:
Let's use proper logging (slf4j) rather than print stack traces like this.

Reply:
Added slf4j logging to this class. Note that the rest of applications do not use self4j, so in a future PR we can extend the use of slf4j to them.